src/gameComponents/DefenseTokens/DefenseToken.java

```diff
@@ -0,0 +1,152 @@
+package gameComponents.DefenseTokens;
+
+import engine.GameConstants;
+
+public class DefenseToken {
+    /**
+    A defense Token is a token that can be assigned to squadrons or ships. A defense token has a variety of states
+    The Defense token class allows for the creation of defense tokens and allows for the various state changes
+    it may undergo during normal use
+     */
+
+    //The type of token
+    protected GameConstants.defenseTokenType type;
+    //The status of the token. All defense tokens start in the ready state
+    protected GameConstants.defenseTokenStatus status = GameConstants.defenseTokenStatus.READY;
+
+    //No Null constructor
+
+    //Typed Constructor
+    public DefenseToken(GameConstants.defenseTokenType type){
+        this.type = type;
+    }
+    //TODO: A String based constructor to use in ship/squadron parsers / factories
+
+    /**
+     * Generic getter method for token status
+     * @return the status of the defense token
+     */
+    public GameConstants.defenseTokenStatus getStatus(){
+        return status;
+    }
+
+    /**
+     * Generic getter for token type
+     * @return the type of defense token
+     */
+    public GameConstants.defenseTokenType getType(){
+        return type;
+    }
+
+    /**
+     * Is the defense token ready?
+     * @return true if token status is ready, else false
+     */
+    public boolean isReady(){
```

Comment:
I'd like to see some tests on this function and the ones following it.  I don't expect anything goofy, but its nice to have in place now so that as this file grows in complexity, we can add more tests and make sure things work as expected.

Reply:
Would like advice on how to proceed with this

Reply:
A few steps to get started.

1) Make a new tile in the tests folder.  On my IDE its green.  Format the file as <class name>Tests.java.

2) Stub out a constructor test by doing something like
@Test
public void testConstructor(){
assertEquals(expected value, getSomething()
}

3) Go ahead and write a test for each function so that we can double check its return value.  Try it a few different ways, like try to exhaust an exhausted token and make sure its false.  

4) If you get stuck you know where to find me.

5) If asserts are being dumb because imports, just grab a similar looking line from the SquadronParserTests